install/install.ps1

```diff
@@ -1,94 +1,203 @@
-cmd.exe /Q /C "mklink /J `"$latestDirectory`" `"$extractedDirectory`"" > $null
-if (!$?) {
-    Write-Error "Linking failed!"
 }
 
-Write-Verbose "Removing ${zipFile}"
-Remove-Item -Force $zipFile
-
-# Permanently add the latest directory to the current user's $PATH (if it's not already there).
-# Note that this will only take effect when a new terminal is started.
-$registryPath = 'Registry::HKEY_CURRENT_USER\Environment'
-$currentPath = (Get-ItemProperty -Path $registryPath -Name PATH -ErrorAction SilentlyContinue).Path
-if ($currentPath -NotMatch 'AzureAuth') {
-    Write-Verbose "Updating `$PATH to include ${latestDirectory}"
-    $newPath = if ($currentPath -Match $null) {
-        "${latestDirectory}"
+function Install-Post-0-4-0 {
+
+    Write-Verbose "Installing using post-0.4.0 method"
+
+    $repo = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_REPO)) { 'AzureAD/microsoft-authentication-cli' } else { $Env:AZUREAUTH_REPO }
+    $releaseName = "azureauth-${version}-win10-x64"
+    $releaseFile = "${releaseName}.zip"
+    $releaseUrl = "https://github.com/${repo}/releases/download/${version}/$releaseFile"
+
+    $azureauthDirectory = if ([string]::IsNullOrEmpty($Env:AZUREAUTH_INSTALL_DIRECTORY)) {
+        ([System.IO.Path]::Combine($Env:LOCALAPPDATA, "Programs", "AzureAuth"))
     } else {
-        "${currentPath};${latestDirectory}"
+        $Env:AZUREAUTH_INSTALL_DIRECTORY
     }
-    Set-ItemProperty -Path $registryPath -Name PATH -Value $newPath
+
+    $targetDirectory = ([System.IO.Path]::Combine($azureauthDirectory, $version))
+    $zipFile = ([System.IO.Path]::Combine($azureauthDirectory, $releaseFile))
+
+    Write-Verbose "Creating ${azureauthDirectory}"
+    $null = New-Item -ItemType Directory -Force -Path $azureauthDirectory
+
+
+    Write-Verbose "Downloading ${releaseUrl} to ${zipFile}"
+    $client = New-Object System.Net.WebClient
+    $client.DownloadFile($releaseUrl, $zipFile)
+
+    # A running instance of azureauth can cause installation to fail, so we try to kill any running instances first.
+    # We suppress taskkill output here because this is a best effort attempt and we don't want the user to see its output.
+    # Here, Get-Process is used to first determine whether there is an existing azureauth process. If there is, kill the existing process first.
+    $ProcessCheck = Get-Process -Name azureauth -ErrorAction SilentlyContinue -ErrorVariable ProcessError
+    if ($null -ne $ProcessCheck) {
+        Write-Verbose "Stopping any currently running azureauth instances"
+        taskkill /f /im azureauth.exe 2>&1 | Out-Null
+
+        # After killing the process it is still possible for there there to be locks on the files it was using (including
+        # its own DLLs). The OS may take an indeterminate amount of time to clean those up, but so far we've observed 1
+        # second to be enough.
+        Start-Sleep -Seconds 1
+    }
+
+    if (Test-Path -Path $targetDirectory) {

```

Comment:
The if-else statement from here might require more commenting or method extraction. I did not totally understand the decisions made here and why.

Reply:
Sure, I will add some comments to help readability. Just wanted to confirm if it is the Test-Path blocks or the NoUpdatePath flag block that's requiring comment?

Reply:
Yes, which part is confusing? 🤔

Reply:
Ah, I see. You just wanted some more explanation from this point onward. I've added some comments. Hope those help.